scripts/deploy-federation.sh

```diff
@@ -128,7 +131,8 @@ to ensure credentials are available for push:
 fi
 
 shift
-JOIN_CLUSTERS="${*}"
+# Allow for no specific JOIN_CLUSTERS: they probably want to kubefed2 themselves.
+JOIN_CLUSTERS="${*-}"
```

Comment:
The original code works for no argument case. I do not understand the concern here, could you please explain it?

Reply:
On my Mac, I have:
```
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
Copyright (C) 2007 Free Software Foundation, Inc.
```

That version of Bash returns: 

```
$ NAMESPACED=y FEDERATION_NAMESPACE=dev bash scripts/deploy-federation.sh quay.io/kubernetes-multicluster/federation-v2:latest
scripts/deploy-federation.sh: line 131: *: unbound variable
$
```

So maybe it's only a problem in older Bash?

Otherwise the script works fine on my Mac.

Reply:
I guess it is a problem in older bash. Linux has no such issue with newer version of bash. I tried on my Mac. It has similar issue.

We have 
```
set -o nounset
```
$* is a special parameter. It seems not defined in old bash but defined in new bash even it has no arguments specified.

As we start building MAC binaries from #684, and the logic here is fine for both platforms. I think it is OK.

Reply:
Thanks!